File: packages/core/src/graph/__tests__/fixtures/sample.rb
class Calculator
  def add(a, b)
    a + b
  end

  def multiply(a, b)
    a * b
  end
end

class AdvancedCalculator < Calculator
  def divide(a, b)
    a / b
  end
end

module MathHelpers
  def square(x)
    x * x
  end
end

File: packages/core/src/graph/__tests__/fixtures/sample.ts
export function greet(name: string): string {
  return `Hello, ${name}!`;
}

export class Calculator {
  add(a: number, b: number): number {
    return a + b;
  }

  multiply(a: number, b: number): number {
    return a * b;
  }
}

export class AdvancedCalculator extends Calculator {
  divide(a: number, b: number): number {
    return a / b;
  }
}
